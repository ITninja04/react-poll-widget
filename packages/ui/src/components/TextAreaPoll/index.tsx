import React from 'react';
import { Form, TextArea } from 'semantic-ui-react';

const TextAreaPoll: React.FC<{

}> = ({

}) => (
<Form>
  <TextArea />
  <
</Form>
);

export default TextAreaPoll;