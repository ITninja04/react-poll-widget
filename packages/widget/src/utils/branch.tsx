export type BranchFunction = (condition: boolean, ComponentIfTrue: React.ReactNode, ComponentIfFalse: React.ReactNode) => React.ReactNode;

export const branch: BranchFunction = (condition, ComponentIfTrue, ComponentIfFalse) => condition
  ? ComponentIfTrue
  : ComponentIfFalse;