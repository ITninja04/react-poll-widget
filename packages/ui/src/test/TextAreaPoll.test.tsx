import TextAreaPoll from "../components/TextAreaPoll";
import { createSnapshotTest } from "./utils";

createSnapshotTest(TextAreaPoll, {}, 'Text area poll')
createSnapshotTest(TextAreaPoll, { loading: true }, 'Loading text area poll')
createSnapshotTest(TextAreaPoll, { error: 'Error' }, 'Text area poll with error')