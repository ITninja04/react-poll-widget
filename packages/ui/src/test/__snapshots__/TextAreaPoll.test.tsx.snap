// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Loading text area poll should render 1`] = `
<DocumentFragment>
  <form
    class="ui loading form"
  >
    <div
      class="field"
    >
      <textarea
        rows="3"
      />
    </div>
    <div
      class="field"
    >
      <button
        class="ui green button"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`(Snapshot) Text area poll should render 1`] = `
<DocumentFragment>
  <form
    class="ui form"
  >
    <div
      class="field"
    >
      <textarea
        rows="3"
      />
    </div>
    <div
      class="field"
    >
      <button
        class="ui green button"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`(Snapshot) Text area poll with error should render 1`] = `
<DocumentFragment>
  <form
    class="ui form"
  >
    <div
      class="error field"
    >
      <textarea
        aria-invalid="true"
        rows="3"
      />
      <div
        aria-atomic="true"
        class="ui pointing above prompt label"
        role="alert"
      >
        Error
      </div>
    </div>
    <div
      class="disabled field"
    >
      <button
        class="ui green disabled button"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;
